src/main/java/org/opentripplanner/routing/impl/ATLFareServiceImpl.java

```diff
@@ -0,0 +1,375 @@
+            this(type, window, maxTransfers, 0, false);
+            if(type.equals(TransferType.TRANSFER_WITH_UPCHARGE)) {
+                throw new IllegalArgumentException("Must specify the upcharge.");
+            }
+        }
+
+        public TransferMeta(TransferType type) {
+            this(type, 0, 0, 0, false);
+            if(!(type.equals(TransferType.END_TRANSFER) || type.equals(TransferType.NO_TRANSFER))) {
+                throw new IllegalArgumentException("Window and maxTransfers arguments required for this transfer type.");
+            }
+        }
+    }
+
+    private static RideType classify(Ride ride) {
+        Route routeData = ride.routeData;
+        String shortName = routeData.getShortName();
+
+        switch(routeData.getId().getAgencyId()) {
+            case COBB_AGENCY_ID:
+                if(shortName.equalsIgnoreCase("100") ||
+                    shortName.equalsIgnoreCase("101") ||
+                    shortName.equalsIgnoreCase("102")) {
+                    return RideType.COBB_EXPRESS;
+                }
+                return RideType.COBB_LOCAL;
+            case XPRESS_AGENCY_ID:
+                return RideType.XPRESS;
+            case MARTA_AGENCY_ID:
+                return RideType.MARTA;
+            case GCT_AGENCY_ID:
+                if(shortName.equalsIgnoreCase("102") ||
+                    shortName.equalsIgnoreCase("103A") ||
+                    shortName.equalsIgnoreCase("110")) {
+                    return RideType.GCT_EXPRESS_Z1;
+                } else if(shortName.equalsIgnoreCase("101") ||
+                    shortName.equalsIgnoreCase("103")) {
+                    return RideType.GCT_EXPRESS_Z2;
+                }
+                return RideType.GCT_LOCAL;
+            case STREETCAR_AGENCY_ID:
+                return RideType.STREETCAR;
+        }
+        // CobbLinc
+        if (ride.feedId.equals("2")) return RideType.COBB_LOCAL;
+
+        // Xpress
+        else if (ride.feedId.equals("6")) {
+            // No need to differentiate between different XPRESS zones because the xfer policies are the same
+            return RideType.XPRESS;
+        }
+
+        // Default to MARTA
+        return RideType.MARTA;
+        // TODO: Finish
+    }
+
+    private static TransferMeta classifyTransfer(RideType toRideType, RideType fromRideType, Fare.FareType fareType) {
+        switch (toRideType) {
+            case STREETCAR:
+                return new TransferMeta(TransferType.NO_TRANSFER);
+            case COBB_LOCAL:
+                if(!isElectronicPayment(fareType)) {
+                    if (fromRideType == RideType.COBB_LOCAL) {
+                        return new TransferMeta(TransferType.FREE_TRANSFER, 180, 4);
+                    }
+                    return new TransferMeta(TransferType.END_TRANSFER);
+                }
+                switch(fromRideType) {
+                    case COBB_LOCAL:
```

Comment:
(and of course, transferring *to* a circulator should not use a transfer b/c there is no transaction upon boarding.

Reply:
This is the same problem with the streetcar, but I had ignored it there since there was no scenario where it could happen. I wasn't aware that CobbLinc has a free bus as well, which means I'll need to come up with a new solution for doing this that looks further back than just the most recent ride.

Reply:
See reply below